index.bs

```diff
@@ -369,6 +448,110 @@ Report long tasks {#report-long-tasks}
+    :: A {{DOMHighResTimeStamp}}, initially 0.
+</dl>
+
+A {{Document}} has a null-or-[=frame timing info=] <dfn>current frame timing info</dfn>, initially null.
+
+
+Report Long Animation Frames {#loaf-processing-model}
+------------------------------------------
+
+<div algorithm="Accessing the frame timing info">
+    To get the <dfn>nearest same-origin root</dfn> for a {{Document}} |document|:
+    1. Let |ancestors| be the [=ancestor navigables=] of |document|.
+
+    1. [=list/For each=] |ancestorNavigable| in |ancestors|:
+        If |ancestorNavigable|'s [=navigable/active document=]'s [=Document/origin=] is [=same origin=] with |document|'s [=Document/origin=],
+            and |ancestorNavigable|'s [=navigable/active document=]'s [=relevant agent=] is |document|'s [=relevant agent=],
+            then return |ancestorNavigable|'s [=navigable/active document=].
+
+    1. Return |document|.
+
+    The <dfn>relevant frame timing info</dfn> for a {{Document}} |document| is its [=nearest same-origin root=]'s [=current frame timing info=].
+</div>
+
+<div algorithm="Report task start time">
+    To <dfn export>report task start time</dfn> given an {{DOMHighResTimeStamp}} |taskStartTime|, and a {{Document}} |document|:
+
+    1. Let |root| be |document|'s [=nearest same-origin root=].
+
+    1. If |root|'s [=current frame timing info=] is null,
+        then set |root|'s [=current frame timing info=] to a new [=frame timing info=] whose [=frame timing info/start time=] is |taskStartTime|.
+
+    1. Set |timingInfo|'s [=current frame timing info=]'s [=frame timing info/current task start time=] to |taskStartTime|.
+
+    1. If |timingInfo|'s [=frame timing info/start time=] is 0, then set |timingInfo|'s [=frame timing info/start time=] to |taskStartTime|.
+</div>
+
+<div algorithm="Report task end time">
+    To <dfn export>report task end time</dfn> given an {{DOMHighResTimeStamp}} |taskEndTime|, and a {{Document}} |document|:
+
+    1. Let |timingInfo| be |document|'s [=relevant frame timing info=].
+
+    1. If |timingInfo| is null, then return.
+
+        Note: This can occur if the browser becomes hidden during the sequence.
+
+    1. Let |safeTaskEndTime| be the [=relative high resolution time=] given |taskEndTime| and |document|'s [=relevant global object=].
+
+    1. Let |safeTaskStartTime| be the [=relative high resolution time=] given |timingInfo|'s [=frame timing info/current task start time=] and |document|'s [=relevant global object=].
+
+    1. Let |currentTaskDuration| be the [=duration=] between |safeTaskStartTime| and |safeTaskEndTime|.
+
+    1. If |currentTaskDuration| is greater than |timingInfo|'s [=frame timing info/longest task duration=], then set |timingInfo|'s [=frame timing info/longest task duration=] to |currentTaskDuration|.
+
+    1. If the user agent believes that updating the rendering of |document|'s [=node navigable=] would have no visible effect, then [=flush frame timing=] given |document| and return.
```

Comment:
Not a fan of "believes" here, but not sure what would be a good replacement..

Reply:
Also, a note explaining why we're flushing here for no-op rendering could be useful

Reply:
The "believes" thing is copied from the equivalent place in the HTML spec

Reply:
That's fine then, I guess..